internal/webhook/webhook.go

```diff
@@ -0,0 +1,193 @@
+// SPDX-FileCopyrightText: SAP SE or an SAP affiliate company and Gardener contributors
+//
+// SPDX-License-Identifier: Apache-2.0
+
+package webhook
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"net/url"
+	"strings"
+
+	"github.com/go-jose/go-jose/v4"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/util/validation/field"
+	certutil "k8s.io/client-go/util/cert"
+	logf "sigs.k8s.io/controller-runtime/pkg/log"
+	"sigs.k8s.io/controller-runtime/pkg/webhook"
+	"sigs.k8s.io/controller-runtime/pkg/webhook/admission"
+
+	authenticationv1alpha1 "github.com/gardener/oidc-webhook-authenticator/apis/authentication/v1alpha1"
+)
+
+var (
+	// log is for logging in this package.
+	log = logf.Log.WithName("openidconnect-resource")
```

Comment:
I wonder if this can be named better. "openidconnect-webhook" does not sound great considering the name of the whole binary.
```suggestion
	log = logf.Log.WithName("openidconnect")
```

Reply:
Sounds good